hw3/hw3-react/src/search/SearchPage.js

```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react';
+import SearchBar from './SearchBar';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Figure from 'react-bootstrap/Figure';
+import FigureImage from 'react-bootstrap/FigureImage';
+import FigureCaption from 'react-bootstrap/FigureCaption';
+
+function SearchPage() {
+  const [results, setResults] = useState([]);
+  const characterMatches = results.map((character) => (
+    <Col>
+      <Figure id={`img-${character.id}`} className="bg-white p-2">
+        <FigureImage
+          width={'250px'}
+          height={'300px'}
+          src={character.imageUrl}
+          alt={`A portrait of ${character.fullName}`}
+        ></FigureImage>
+        <FigureCaption>{character.fullName}</FigureCaption>
+        <small className="d-block">{character.title}</small>
+      </Figure>
+    </Col>
+  ));
+
+  function searchName(name) {
+    // this isn't working
+    setResults([]);
+    alert(`You searched for ${name}!`);
+    const storedCharacters = sessionStorage.getItem('characters');
```

Comment:
Although sessionStorage does retrieve the data information, I would consider passing the data that you receive from the API call to the SearchPage component so that this wouldn't be required. It aligns more with how React should function in the sense that you pass data to other components to use.

Reply:
That sounds like a much more valid solution, thanks! I'll have to look into refactoring this portion of the code.